Language/Expressions/Constants/equals_expression_t03.dart

```diff
@@ -9,22 +9,21 @@
 /// • An expression of one of the forms e1 == e2 or e1 != e2 where e1 and e2 are
 ///   constant expressions that evaluate to a numeric, string or boolean value
 ///   or to null.
-/// @description Checks that it is a compile-time error when an expression of
-/// the form e1 == e2 where e1 does not evaluate to a numeric, string or boolean
-/// value, is used to initialize a constant variable.
+/// @description Checks that it is not an time error when an expression of
+/// the form e1 == e2 or e1 != e2 where e1 does not evaluate to a numeric,
+/// string or boolean value, is used to initialize a constant variable.
 /// @author kaigorodov
 
+// SharedOptions=--enable-experiment=patterns
+
 class A {
   const A() : a = "hello";
   final a;
 }
 
 final constList = const [
-//                ^
-// [cfe] unspecified
-   const A() == "hello",
-// ^
-// [analyzer] unspecified
+  const A() == "hello",
+  const A() != "world"

```

Comment:
To fit the description, we might as well have a couple of cases like `1.5 == A()` and `B() == null`, where `B` is a class that does not have primitive equality (because _any_ constant is OK when the right operand is `null`).